apps/src/templates/levelSummary/FreeResponseAIEvaluation.tsx

```diff
@@ -0,0 +1,128 @@
+  FreeResponseAIEvaluationProps
+> = ({responses, levelInstructions}) => {
+  const [evaluationsPending, setEvaluationsPending] = useState<boolean>(false);
+  const [evaluations, setEvaluations] = useState<StudentResponse[]>([]);
+  const [evaluationCount, setEvaluationCount] = useState<number>(0);
+  const [aiSummary, setAiSummary] = useState<string>('');
+  const evaluationComplete = responses.length === evaluationCount;
+
+  useEffect(() => {
+    if (evaluationComplete) {
+      setEvaluationsPending(false);
+      summarizeStudentEvaluations(evaluations);
+    }
+  }, [evaluations, evaluationComplete]);
+
+  const basePrompt =
+    'You are a teaching assistant for a high school AP Computer Science class where the students are learniing JavaScript.';
+
+  const getAIEvaluations = async () => {
+    setEvaluationsPending(true);
+    const responsePromises = responses.map(async studentResponse => {
+      return evaluateStudentResponse(studentResponse);
+    });
+
+    await Promise.allSettled(responsePromises);
+  };
+
+  const evaluateStudentResponse = async (studentResponse: StudentResponse) => {
+    const studentPrompt = `${basePrompt} Please review the student's responses and indicate whether the response is "great", "ok", or "needs revision". Provide one sentence with your reasoning. The student's instructions are: ${levelInstructions}.`;
+    const studentResponseString = `${studentResponse.student_display_name} replied ${studentResponse.text}`;
+    const chatApiResponse = await getChatCompletionMessage(
+      studentResponseString,
+      [],
+      studentPrompt
+    );
+    const aiEvaluation = chatApiResponse.assistantResponse;
+    if (aiEvaluation) {
+      const evaluation = {
+        ...studentResponse,
+        aiEvaluation: aiEvaluation,
+      };
+      setEvaluations(prevEvaluations => [...prevEvaluations, evaluation]);
+      setEvaluationCount(prevCount => prevCount + 1);
+    }
+  };
+
+  const summarizeStudentEvaluations = async (
+    evaluations: StudentResponse[]
+  ) => {
+    const sectionPrompt = `${basePrompt} Please review the evaluations of the student responses and based on the results indicate whether the teacher should "review the concept" or "move on to the next lesson". Provide one sentence with your reasoning.`;
+    const chatApiResponse = await getChatCompletionMessage(
+      evaluations.map(evaluation => evaluation.aiEvaluation).join(' '),
+      [],
+      sectionPrompt
+    );
+    const summary = chatApiResponse.assistantResponse;
+    if (summary) {
+      setAiSummary(summary);
+    } else {
+      setAiSummary('Uh oh!');
+    }
+  };
+
+  return (
+    <div>
+      <h2>AI Analysis (prototype)</h2>
+      <Button
+        text="Evaluate student responses"
+        onClick={getAIEvaluations}
+        disabled={!responses.length || evaluationsPending}
+        isPending={evaluationsPending}
+      />
+      {evaluationComplete && aiSummary && (
+        <div>
+          <br />
```

Comment:
Similar concern here ... I think we'd use CSS to style here instead of inserting a line break (unless I'm missing something).

This isn't blocking, just wondering out loud about the line between prototype and "code we'll build off of" 😀

Reply:
interesting discussion! I would suggest
1. allow shortcuts in this PR since it is a prototype, as long as it won't break existing users
2. clearly define what is prototype code and what isn't
3. come up with your own plan for how to ensure prototype code gets re-reviewed before being shipped

perhaps it would suffice to have a TODO in each file that will need to go through review again before being shipped to end users?

Reply:
In the AI Tutor early days I made "Future Work" doc which was easy to reference when it came time to pick up that project after it was paused. Starting [something similar for Measures](https://docs.google.com/document/d/1P53DZhMELqrt0VArMsqVnN-WpCOLR7-R4mt4bBtMlGs/edit?tab=t.0)